infer.py

```diff
@@ -1,85 +1,138 @@
+
+    parser.add_argument(
+        "--render_every",
+        type=int,
+        default=2,
+        help="how many frames to skip between renders",
+    )
+    parser.add_argument(
+        "--video_length",
+        type=float,
+        default=20.0,
+        help="desired length of video in seconds",
+    )
     parser.add_argument("--width", type=int, default=640, help="width of the video frame")
     parser.add_argument("--height", type=int, default=480, help="height of the video frame")
+    parser.add_argument(
+        "--env_module",
+        type=str,
+        default="environment",
+        help="Name of the environment module to import.",
+    )
     args = parser.parse_args()
 
-    env = HumanoidEnv()
+    env_module = importlib.import_module(args.env_module)
+
+    env = env_module.HumanoidEnv()
     rng = jax.random.PRNGKey(0)
 
-    actor_params, _ = load_models(args.actor_path, args.critic_path)
-    actor = Actor(input_size=env.observation_size, action_size=env.action_size, key=rng)
+    model_path = f"models/{args.env_name}_model.pkl"
+    video_path = f"videos/{args.env_name}_video.mp4"
+    print("Loading model from", model_path)
+    loaded_params = load_model(model_path)
+    network = ActorCritic(env.action_size, activation="tanh")
 
     reset_fn = jax.jit(env.reset)
     step_fn = jax.jit(env.step)
 
     fps = int(1 / env.dt)
     max_frames = int(args.video_length * fps)
     rollout: list[Any] = []
+    episode_reward = 0
+    total_reward = 0
+    rng, reset_rng = jax.random.split(rng)
+    state = reset_fn(reset_rng)
 
-    for episode in range(args.num_episodes):
-        rng, reset_rng = jax.rand.split(rng)
-        state = reset_fn(reset_rng)
-        obs = state.obs
+    episodes = 0
 
-        total_reward = 0
+    for step in range(max_frames):
+        obs = state.obs
+        rollout.append(state.pipeline_state)
 
-        for step in range(args.max_steps):
-            if len(rollout) < max_frames:
-                rollout.append(state.pipeline_state)
+        rng, _rng = jax.random.split(rng)
+        pi, _ = network.apply(loaded_params, obs)
+        action = pi.sample(seed=_rng)
 
-            action = choose_action(actor_params, obs, actor)
-            state = step_fn(state, action)
-            obs = state.obs
-            total_reward += state.reward
+        rng, _rng = jax.random.split(rng)
+        state = step_fn(state, action, _rng)
 
-            if state.done:
-                break
+        total_reward += state.reward
+        episode_reward += state.reward
 
-        print(f"Episode {episode + 1} total reward: {total_reward}")
+        if state.done:
+            episodes += 1
+            print("Episode", episodes, "reward:", episode_reward)
+            episode_reward = 0
 
         if len(rollout) >= max_frames:
             break
 
-    # Trim rollout to desired length
-    rollout = rollout[:max_frames]
+    # def scan_env(carry, t):
```

Comment:
Why commenting out?

Reply:
i had scan env setup for vecotrized environments in infer -- should prob make scan env setup for normal environments in infer though you're right. hasn't been a problem so far b/c training is enough time that waiting a little longer to see the video isn't too bad